miden-tx/src/host/mod.rs

```diff
@@ -321,7 +339,35 @@ impl<A: AdviceProvider> Host for TransactionHost<A> {
         process: &S,
         injector: AdviceInjector,
     ) -> Result<HostResponse, ExecutionError> {
-        self.adv_provider.set_advice(process, &injector)
+        match injector {
+            AdviceInjector::SigToStack { .. } => {
+                let pub_key = process.get_stack_word(0);
+                let msg = process.get_stack_word(1);
+                let signature_key = Hasher::hash_elements(&[pub_key, msg].concat());
+
+                let signature =
+                    if let Some(signature) = self.adv_provider.get_mapped_values(&signature_key) {
+                        signature.to_vec()
+                    } else {
+                        let account_delta = self.account_delta.clone().into_delta();
+
+                        let signature: Vec<Felt> = self
+                            .tx_authenticator
+                            .get_signature(pub_key, msg, &account_delta)
+                            .map_err(|err| ExecutionError::EventError(err.to_string()))?;
```

Comment:
This is not really an `EvenError` but I didn't find one that suited it better

Reply:
I totally missed that (although it's `&'static str` which is a bit more inconvenient).

Reply:
Probably worth creating an issue (or maybe even a small PR) in `miden-vm` to change it to `String`. We won't be able to use it until the next minor release of Miden VM (which won't happen until June for sure) - but would be useful further down the line.